packages/notification-email-sender/src/services/notificationEmailSenderService.ts

```diff
@@ -0,0 +1,634 @@
+          interopFeUrl: `https://${interopFeBaseUrl}/ui/it/erogazione/richieste/${agreement.id}`,
+          producerName: producer.name,
+          consumerName: consumer.name,
+          eserviceName: eservice.name,
+          eserviceVersion: descriptor.version,
+          activationDate,
+        }),
+      };
+      try {
+        logger.info(
+          `Sending email for agreement ${agreement.id} activation (${sesEmailManager.kind})`
+        );
+        await sesEmailManager.send(mail.from, mail.to, mail.subject, mail.body);
+        logger.info(
+          `Email sent for agreement ${agreement.id} activation (${sesEmailManager.kind})`
+        );
+      } catch (err) {
+        logger.error(
+          `Unexpected error sending email for agreement ${agreement.id} activation (${sesEmailManager.kind}): ${err}`
+        );
+        throw genericInternalError(
+          `Error sending email for agreement ${agreement.id}: ${err}`
+        );
+      }
+    },
+    sendAgreementRejectedEmail: async (
+      agreementV2Msg: AgreementV2,
+      logger: Logger
+    ) => {
+      const agreement = fromAgreementV2(agreementV2Msg);
+
+      const [htmlTemplate, eservice, producer, consumer] = await Promise.all([
+        retrieveHTMLTemplate(
+          eventMailTemplateType.agreementRejectedMailTemplate
+        ),
+        retrieveAgreementEservice(agreement, readModelService),
+        retrieveTenant(agreement.producerId, readModelService),
+        retrieveTenant(agreement.consumerId, readModelService),
+      ]);
+
+      const rejectionDate = getFormattedAgreementStampDate(
+        agreement,
+        "rejection"
+      );
+
+      const consumerEmail = getLatestTenantMailOfKind(
+        consumer.mails,
+        tenantMailKind.ContactEmail
+      );
+
+      if (!consumerEmail) {
+        logger.warn(
+          `Consumer email not found for agreement ${agreement.id}, skipping email`
+        );
+        return;
+      }
+
+      const mail = {
+        from: { name: sesSenderData.label, address: sesSenderData.mail },
+        subject: `Richiesta di fruizione per ${eservice.name} rifiutata`,
+        to: [consumerEmail.address],
+        body: templateService.compileHtml(htmlTemplate, {
+          interopFeUrl: `https://${interopFeBaseUrl}/ui/it/erogazione/richieste/${agreement.id}`,
+          producerName: producer.name,
+          consumerName: consumer.name,
+          eserviceName: eservice.name,
+          rejectionDate,
+        }),
+      };
+
+      try {
+        logger.info(`Sending email for agreement ${agreement.id} rejection`);
+        await sesEmailManager.send(mail.from, mail.to, mail.subject, mail.body);
+        logger.info(`Email sent for agreement ${agreement.id} rejection`);
+      } catch (err) {
+        logger.warn(
+          `Error sending email for agreement ${agreement.id}: ${err}`
+        );
+        throw genericInternalError(
+          `Error sending email for agreement ${agreement.id}: ${err}`
+        );
```

Comment:
Double-check that you specify the operation (if needed) also in all error logs of this file

```suggestion
        logger.warn(
          `Error sending email for agreement ${agreement.id} rejection: ${err}`
        );
        throw genericInternalError(
          `Error sending email for agreement ${agreement.id} rejection: ${err}`
        );
```